tools/gradle/README.md

```diff
@@ -0,0 +1,193 @@
+# Ice Slice Tools Gradle Plugin
+
+The **Ice Slice Tools Gradle Plugin** integrates the **Slice-to-Java compiler (`slice2java`)** into Gradle-based
+**Java** and **Android** projects. It automates the compilation of Slice (`.ice`) files into Java source files and
+integrates seamlessly with the Gradle build system.
+
+## Features
+
+- **Automatic Slice Compilation**: Compiles Slice definitions in `.ice` files into Java `.java` files using
+`slice2java`.
+- **Supports Both Java and Android Projects**: Integrates with the Gradle Java and Android plugins.
+- **Incremental Compilation**: Compiles only modified `.ice` files.
+- **Automatic Dependency Management**: Detects and handles Slice file dependencies.
+- **Customizable Compiler Arguments**: Supports `-I`, `-D`, and other `slice2java` options.
+- **Automatic Slice Compiler Extraction**: The plugin includes the Slice for Java compiler, `slice2java`, with binaries
+for Linux (x64 and arm64), macOS (x64 and arm64), and Windows (x64).
+
+## Installation
+
+### Applying the Plugin
+
+Apply the plugin to your `build.gradle` (Groovy) or `build.gradle.kts` (Kotlin):
+
+**Groovy DSL:**
+
+```groovy
+plugins {
+    id 'com.zeroc.ice.slice-tools' version '3.8.0-alpha.0'
+}
+```
+
+**Kotlin DSL:**
+
+```kotlin
+plugins {
+    id("com.zeroc.ice.slice-tools") version "3.8.0-alpha.0"
+}
+```
+
+## Configuration
+
+The plugin provides the `slice` extension for configuring Slice compilation settings.
+
+### Slice Extension Properties
+
+| Property Name       | Description                                                                                                   |
+|---------------------|---------------------------------------------------------------------------------------------------------------|
+| `toolsPath`         | Path to the directory containing the `slice2java` compiler. If not set, the plugin uses the bundled compiler. |
```

Comment:
Do we need this toolsPath escape?

I feel we should always use the slice2java in the plugin. And we want another slice2java, just build your local plugin (which should be easy to do).

Reply:
This is similar to what we do with MSBuild, we don't publish NuGet packages and consume them.

I think the extra step to first publish a plugin for local use would add complexity to our builds.

* We have to build slice2java
* Copy it to the plugin resource folder
* Publish the plugin to a local maven repository